### ISSUE TITLE
The code of converting the depth map into a point cloud image

### ISSUE THREAD

**Issue by LP940708** [YvanYin/DiverseDepth#14]
Hello author, I converted the depth map into a point cloud image. After converting with my code, the effect image visualized by meshlab is very different from the effect image in your paper. Could you share your code for converting depth map to point cloud map?
The following is my code:
----------------------------------------------------------------------------------------------------------------------------------------------
from PIL import Image
import pandas as pd
import numpy as np
import time
import open3d as o3d

class point_cloud_generator():

    def __init__(self, rgb_file, depth_file, pc_file, focal_length, scalingfactor):
        self.rgb_file = rgb_file
        self.depth_file = depth_file
        self.pc_file = pc_file
        self.focal_length = focal_length
        self.scalingfactor = scalingfactor
        self.rgb = Image.open(rgb_file)
        self.depth = Image.open(depth_file).convert('I')
        self.width = self.rgb.size[0]
        self.height = self.rgb.size[1]

    def calculate(self):
        t1=time.time()
        depth = np.asarray(self.depth).T
        self.Z = depth / self.scalingfactor
        X = np.zeros((self.width, self.height))
        Y = np.zeros((self.width, self.height))
        for i in range(self.width):
            X[i, :] = np.full(X.shape[1], i)

        self.X = ((X - self.width / 2) * self.Z) / self.focal_length
        for i in range(self.height):
            Y[:, i] = np.full(Y.shape[0], i)
        self.Y = ((Y - self.height / 2) * self.Z) / self.focal_length

        df=np.zeros((6,self.width*self.height))
        df[0] = self.X.T.reshape(-1)
        df[1] = -self.Y.T.reshape(-1)
        df[2] = -self.Z.T.reshape(-1)
        img = np.array(self.rgb)
        df[3] = img[:, :, 0:1].reshape(-1)
        df[4] = img[:, :, 1:2].reshape(-1)
        df[5] = img[:, :, 2:3].reshape(-1)
        self.df=df
        t2=time.time()
        print('calcualte 3d point cloud Done.',t2-t1)

    def write_ply(self):
        t1=time.time()
        float_formatter = lambda x: "%.4f" % x
        points =[]
        for i in self.df.T:
            points.append("{} {} {} {} {} {} 0\n".format
                          (float_formatter(i[0]), float_formatter(i[1]), float_formatter(i[2]),
                           int(i[3]), int(i[4]), int(i[5])))

        file = open(self.pc_file, "w")
        file.write('''ply
        format ascii 1.0
        element vertex %d
        property float x
        property float y
        property float z
        property uchar red
        property uchar green
        property uchar blue
        property uchar alpha
        end_header
        %s
        ''' % (len(points), "".join(points)))
        file.close()
        t2=time.time()
        print("Write into .ply file Done.",t2-t1)

a = point_cloud_generator('rgb1.png', 'depth1.png', '1.ply',focal_length=50, scalingfactor=1)
a.calculate()
a.write_ply()

----------------------------------------------------------------------------------------------------------------------------------------------
![1](https://user-images.githubusercontent.com/52646728/94529241-0e071200-026c-11eb-805e-cff74b5f64fc.png)
![11](https://user-images.githubusercontent.com/52646728/94529265-165f4d00-026c-11eb-99cb-693888393204.png)
![point](https://user-images.githubusercontent.com/52646728/94529267-18291080-026c-11eb-8784-c296bc9acc9c.png)

**Comment by YvanYin** [comment]
Hi,  as there is an unknow shift in the depth map. Therefore, if you want to reconstruct a more reasonable shape, you should add a shift on the depth map before the reconstruction. For example, depth /= depth.max(). depth += 5.

**Comment by LP940708** [comment]
Hello author, I saw that the picture of the elephant above in your paper comes from the DIW data set. I downloaded the DIW data set, but I found that the DIW data set did not indicate the focal length of each picture, but in your paper, the effect of reconstructing the point cloud image from the depth map for this elephant image is very good, so I adjusted the focal length of this elephant image and finally constructed a good point cloud image (focal length=600.0, scalingFactor = 1.0 ). In your paper, if you don’t know the focal length, did you set a virtual focal length to finally reconstruct the point cloud image?

The following is the point cloud image I reconstructed(focal length=600.0, scalingFactor = 1.0):
![snapshot00](https://user-images.githubusercontent.com/52646728/95155105-009adc00-07c6-11eb-9337-f802090a7657.png)
